### PR TITLE
ref: Deprecate get_legacy_message

### PR DESCRIPTION
**Pull request by lynnagara**
Move the implementation into the Pagerduty plugin since this is the only
place it is used.

### CODE REVIEW

File: src/sentry_plugins/pagerduty/plugin.py [hunk 1]
```diff
@@ -49,7 +49,10 @@ def notify_users(self, group, event, fail_silently=False, **kwargs):
         if not self.is_configured(group.project):
             return
 
-        description = event.get_legacy_message()[:1024]
+        # TODO: This should eventually be event.title in line with other plugins.
+        # However, we should notify users first, since PD alert routing may be
+        # based off the message field.
+        description = (event.real_message or event.message)[:1024]
```
Comment (fpacifici) [inline src/sentry_plugins/pagerduty/plugin.py]: What's the difference in content between the logentry attribute you were using before and real_message/event.message

Comment (lynnagara) [reply src/sentry_plugins/pagerduty/plugin.py]: There shouldn't be any difference in content here, this is just a refactor to clean up the event class and move the Pagerduty specific message into the plugin

Comment (fpacifici) [reply src/sentry_plugins/pagerduty/plugin.py]: Could you please expand on the "shouldn't be any difference" ? You are now fetching a different field from the event. Where was the old `logentry` populated from ?

Comment (lynnagara) [reply src/sentry_plugins/pagerduty/plugin.py]: I mean if you look at the implementation of `real_message` vs `get_legacy_message()`, they are almost exactly the same except that legacy message falls back to `event.message`, which is what we have done here.

Comment (fpacifici) [reply src/sentry_plugins/pagerduty/plugin.py]: damn you are right, I was looking at the wrong method.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**